rp2040-hal/src/clocks/mod.rs

```diff
@@ -301,6 +302,22 @@ impl ClocksManager {
             .configure_clock(&self.system_clock, self.system_clock.freq())
     }
 
+    /// Configure the clocks staying ON during deep-sleep.
+    pub fn configure_sleep_en<F0, F1>(&mut self, sleep_en0: F0, sleep_en1: F1)
```

Comment:
I'm not fond of it either.
I prefer this to taking a u32 bit pattern because in its use place it is self explanatory.

I think the "real" good solution would be to implement a bit field with accessors similar to what svd2rust generates.
I was being lazy yesterday :)

Reply:
The latest commit should fix this.
It introduces a new dependency which has itself none and is fairly stable.